lib/dialects/mariadb/query.js

```diff
@@ -0,0 +1,358 @@
+'use strict';
+
+const Utils = require('../../utils');
+const logger = require('../../utils/logger');
+const debug = logger.getLogger().debugContext('sql:mariadb');
+const AbstractQuery = require('../abstract/query');
+const uuidv4 = require('uuid/v4');
+const sequelizeErrors = require('../../errors');
+const _ = require('lodash');
+const DataTypes = require('../../data-types');
+
+const ER_DUP_ENTRY = 1062;
+const ER_ROW_IS_REFERENCED = 1451;
+const ER_NO_REFERENCED_ROW = 1452;
+
+class Query extends AbstractQuery {
+  constructor(connection, sequelize, options) {
+    super();
+    this.connection = connection;
+    this.instance = options.instance;
+    this.model = options.model;
+    this.sequelize = sequelize;
+    this.uuid = uuidv4();
+    this.options = Object.assign({
+      logging: console.log,
+      plain: false,
+      raw: false,
+      showWarnings: false
+    }, options || {});
+
+    this.checkLoggingOption();
+  }
+
+  static formatBindParameters(sql, values, dialect) {
+    const bindParam = [];
+    const replacementFunc = (match, key, val) => {
+      if (val[key] !== undefined) {
+        bindParam.push(val[key]);
+        return '?';
+      }
+      return undefined;
+    };
+    sql = AbstractQuery.formatBindParameters(sql, values, dialect,
+      replacementFunc)[0];
+    return [sql, bindParam.length > 0 ? bindParam : undefined];
+  }
+
+  run(sql, parameters) {
+    this.sql = sql;
+
+    //do we need benchmark for this query execution
+    const benchmark = this.sequelize.options.benchmark
+      || this.options.benchmark;
+    const showWarnings = this.sequelize.options.showWarnings
+      || this.options.showWarnings;
+
+    let queryBegin;
+    if (benchmark) {
+      queryBegin = Date.now();
+    } else {
+      this.sequelize.log(
+        `Executing (${this.connection.threadId || 'default'}): ${this.sql}`,
```

Comment:
`this.connection.uuid` this is useful to determine which logs are running under transaction, please use this format for other logs in this file as well